extensions/standard-processors/processors/RetryFlowFile.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "RetryFlowFile.h"
+
+#include "core/PropertyValidation.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+core::Property RetryFlowFile::RetryAttribute(core::PropertyBuilder::createProperty("Retry Attribute")
+    ->withDescription(
+        "The name of the attribute that contains the current retry count for the FlowFile."
+        "WARNING: If the name matches an attribute already on the FlowFile that does not contain a numerical value, "
+        "the processor will either overwrite that attribute with '1' or fail based on configuration.")
+    ->withDefaultValue("flowfile.retries")
```

Comment:
Shouldn't this be required?

Reply:
it might be beneficial to make all new properties required by default, and if in the future it proves to be a pain point, we could always give a default based on real usage requirements, while if we would want to make something required later we would break previous flows

Reply:
The NiFi implementation has this default value set:
http://nifi.apache.org/docs/nifi-docs/components/org.apache.nifi/nifi-standard-nar/1.11.4/org.apache.nifi.processors.standard.RetryFlowFile/index.html

Reply:
Looking at NiFi code is always a good source of hints and what we see there worth consideration, but it's not an unerring guidance or revelation. 
In case we think that our implementation definitely requires a valid value for the given property to work, I think we should make it required.

In case of properties that define attribute names I would also consider something like this: https://www.javadoc.io/doc/org.apache.nifi/nifi-processor-utils/0.7.0/org/apache/nifi/processor/util/StandardValidators.html#NON_BLANK_VALIDATOR
As we request a valid string here that can name an attribute.

Reply:
Added non-blank validator.